src/lib.rs

```diff
@@ -1,3 +1,38 @@
+//! Experimental hash table implementation in just Rust (stable, no unsafe
+//! code).
+//!
+//! This is inpired by Python 3.6's new dict implementation (which remembers
+//! the insertion order and is fast to iterate, and is compact in memory).
+//!
+//! This implementation corresponds to their "compact hash map" idea as it is
+//! now, and has consistent order.
```

Comment:
Thanks for the PR. I understand that the crate is confusingly documented as it is, but rustdoc is where it should be correct. The ordering explanation on the struct OrderMap itself is better and more correct. Resting on the reference to Python makes it easy to misunderstand what it actually does.